Sources/BraveWallet/Extensions/BraveWalletSwiftUIExtensions.swift

```diff
@@ -74,6 +78,37 @@ extension BraveWallet.NetworkInfo: Identifiable {
     guard let logo = nativeTokenLogo else { return nil }
     return UIImage(named: logo, in: .module, with: nil)
   }
+  
+  public var networkLogo: String? {
+    if chainId.caseInsensitiveCompare(BraveWallet.MainnetChainId) == .orderedSame {
+      return "eth-asset-icon"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.SolanaMainnet) == .orderedSame {
+      return "sol-asset-icon"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.FilecoinMainnet) == .orderedSame {
+      return "filecoin-asset-icon"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.PolygonMainnetChainId) == .orderedSame {
+      return "matic"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.BinanceSmartChainMainnetChainId) == .orderedSame {
+      return "bnb-asset-icon"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.CeloMainnetChainId) == .orderedSame {
+      return "celo"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.AvalancheMainnetChainId) == .orderedSame {
+      return "avax"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.FantomMainnetChainId) == .orderedSame {
+      return "fantom"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.AuroraMainnetChainId) == .orderedSame {
+      return "aurora"
+    } else if chainId.caseInsensitiveCompare(BraveWallet.OptimismMainnetChainId) == .orderedSame {
+      return "optimism"
+    } else {
+      return iconUrls.first
+    }
+  }
```

Comment:
Right now these strings are duplicated in `BraveWallet.NetworkInfo.networkLogo`, `BraveWallet.NetworkInfo.nativeTokenLogo`, and in `NetworkIcon.networkImageInfo`. We should avoid having this much duplicated code, especially strings that won't get caught by the compiler. 

Maybe we could create an enum for each image string? That way we don't miss updating any strings as they're all hardcoded in one location. We might even be able to move these computed properties into the enum, or make them functions in the enum (or can we reduce so only one property is needed total?)?

ex.
```
enum AssetImageName: String {
  case solana = "sol-asset-icon"
  <...>
  
  func networkLogoImageName(for chainId: String) -> String? {
    switch chainId {
    case .solana: return AssetImageName.solana.rawValue
    default: return nil
    }
  }
}
```

Maybe there is another way to reduce the duplicated code if these 2 properties can't be merged into one?

Reply:
updated in [188d673](https://github.com/brave/brave-ios/pull/7361/commits/188d6732e73af0f5cf9271d747aef6b7aedb266b)

Reply:
Can we merge these properties? Can we reduce the duplicate code at all? SonarCloud is also flagging the duplicate code (I think it will flag it every time we touch this file).